R/zchunk_L2042.resbio_input_irr_mgmt.R

```diff
@@ -36,96 +54,294 @@ module_aglu_L2042.resbio_input_irr_mgmt_DISABLED <- function(command, ...) {
-    L2041.AgResBioCurve_For <- get_data(all_data, "L2041.AgResBioCurve_For")
-    L2041.StubResBioCurve_Mill <- get_data(all_data, "L2041.StubResBioCurve_Mill")
-    L2041.AgResBioCurve_ag_irr <- get_data(all_data, "L2041.AgResBioCurve_ag_irr")
-
-    # ===================================================
-    # TRANSLATED PROCESSING CODE GOES HERE...
-    #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    basin_to_country_mapping <- get_data(all_data, "water/basin_to_country_mapping")
+    A_demand_technology <- get_data(all_data, "aglu/A_demand_technology")
+    A_resbio_curves <- get_data(all_data, "aglu/A_resbio_curves")
+    A_bio_frac_prod_R <- get_data(all_data, "aglu/A_bio_frac_prod_R")
+    L111.ag_resbio_R_C <- get_data(all_data, "L111.ag_resbio_R_C")
+    L103.ag_Prod_Mt_R_C_Y_GLU <- get_data(all_data, "L103.ag_Prod_Mt_R_C_Y_GLU")
+    L123.For_Prod_bm3_R_Y_GLU <- get_data(all_data, "temp-data-inject/L123.For_Prod_bm3_R_Y_GLU")
+
+    # following lines temporary while L123 is temp-data-inject
+    L123.For_Prod_bm3_R_Y_GLU %>%
+      gather(year, prod, -GCAM_region_ID, -GCAM_commodity, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L123.For_Prod_bm3_R_Y_GLU
+
+
+    # following line temproary while L103 is output from zchunk_LA103.ag_R_C_Y_GLU.R as grouped
+    L103.ag_Prod_Mt_R_C_Y_GLU %>% ungroup -> L103.ag_Prod_Mt_R_C_Y_GLU
+
+    # the following lines convert basin identification from the current GLU### level 1 names to the
+    # level 2 names.
+    L103.ag_Prod_Mt_R_C_Y_GLU %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L103.ag_Prod_Mt_R_C_Y_GLU
+
+    L123.For_Prod_bm3_R_Y_GLU %>%
+      replace_GLU(map = basin_to_country_mapping) ->
+      L123.For_Prod_bm3_R_Y_GLU
+
+    # add actual region names to table A_bio_frac_prod_R for joining later
+    A_bio_frac_prod_R %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") ->
+      A_bio_frac_prod_R
+
+
+    # Forestry and Mill residue bio
+
+    # The function, add_bio_res_params_For_Mill, takes a data frame and adds user specified parameters and
+    # then repeats the resulting data frame for all MODEL_YEARS to form the  AgResBio_source
+    # output table
+    add_bio_res_params_For_Mill <- function(df, residueBiomassProduction, massConversion, harvestIndex, erosCtrl, massToEnergy, waterContent){
```

Comment:
Maybe have defaults `residueBiomassProduction = aglu.AVG_WOOD_DENSITY_KGM3`, etc., to make code below cleaner.